launcher/ui/widgets/EnvironmentVariables.cpp

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (C) 2023 TheKodeToad <TheKodeToad@proton.me>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <QKeyEvent>
+
+#include "Application.h"
+#include "EnvironmentVariables.h"
+#include "ui/dialogs/CustomMessageBox.h"
+#include "ui_EnvironmentVariables.h"
+
+EnvironmentVariables::EnvironmentVariables(QWidget* parent) : QWidget(parent), ui(new Ui::EnvironmentVariables)
+{
+    ui->setupUi(this);
+    ui->list->installEventFilter(this);
+
+    ui->list->sortItems(0, Qt::AscendingOrder);
+    ui->list->setSortingEnabled(true);
+    ui->list->header()->resizeSections(QHeaderView::Interactive);
+    ui->list->header()->resizeSection(0, 200);
+
+    connect(ui->add, &QPushButton::clicked, this, [this] {
+        auto item = new QTreeWidgetItem(ui->list);
+        item->setText(0, "ENV_VAR");
+        item->setText(1, "value");
+        item->setFlags(item->flags() | Qt::ItemIsEditable);
+        ui->list->addTopLevelItem(item);
+        ui->list->selectionModel()->select(ui->list->model()->index(ui->list->indexOfTopLevelItem(item), 0),
+                                           QItemSelectionModel::ClearAndSelect | QItemSelectionModel::SelectionFlag::Rows);
+        ui->list->editItem(item);
+    });
+
+    connect(ui->remove, &QPushButton::clicked, this, [this] {
+        for (QTreeWidgetItem* item : ui->list->selectedItems())
+            ui->list->takeTopLevelItem(ui->list->indexOfTopLevelItem(item));
+    });
+
+    connect(ui->clear, &QPushButton::clicked, this, [this] { ui->list->clear(); });
+
+    connect(ui->globalOverride, &QCheckBox::clicked, this, [this](bool state) {
+        if (!state)
+            return;
+
+        auto global = APPLICATION->settings()->get("Env").toMap();
+        if (global.isEmpty())
+            return;
+
+        auto response = CustomMessageBox::selectable(
+                            this, tr("Reset"),
+                            tr("You have chosen to ignore global settings.\n\nWould you like to clear the current variables and copy "
```

Comment:
Shouldn't this be the default? i.e. if a user enables the override in the first place, it should just copy the global settings. At least that's how all other overrides work currently

Reply:
I feel like you wouldn't want that in most cases

Reply:
I also would prefer to be asked what to do.
In most places where the reset happens are just one-line settings so that's okay as you can change it back easily.
(*)It would be nice to have a similar popup for the java args

Reply:
hmm, i could change it but environment variables don't normally work like this - there's already a lot of inherited environment variables, right?

Reply:
+1 this makes a lot more sense as there wouldn't be any question as to what env vars are being applied to the game - plus like scrump said, exporting is also an important factor

Reply:
It will be inconsistent between systems :woman_shrugging: 

You might run the executable with an env variable anyway